### ISSUE TITLE
[build] Dependabot does not update .csproj files in templates area.

### ISSUE THREAD

**Issue by kaby76** [antlr/grammars-v4#3480]
The Dependabot program is [specified to run on the CSharp templates directory](https://github.com/antlr/grammars-v4/blob/4ee2f74c11722d0b7d6c52cc6a86d7f2564fe11c/.github/dependabot.yml#L19), but I've never once seen a PR created for modifying the [line for antlr4.runtime.standard in the .csproj file in the templates](https://github.com/antlr/grammars-v4/blob/4ee2f74c11722d0b7d6c52cc6a86d7f2564fe11c/_scripts/templates/CSharp/Test.csproj#L15). We know that [antlr4.runtime.standard](https://www.nuget.org/packages?q=antlr4.runtime.standard) was recently released. 

It's likely that Dependabot doesn't parse the .csproj file in template form because the open angle bracket `<` in `<PackageReference Include="Antlr4.Runtime.Standard" Version ="4.13.0">` needs to be escaped as `\<` for StringTemplate. I may be able to change the delimiter characters in StringTemplate to something that doesn't conflict with the Dependabot parser of .csproj files. [This constructor call](https://github.com/kaby76/Domemtech.Trash/blob/b30d2c05794a4e6245032f03bd888b176322d750/src/trgen/Command.cs#L1594) will need to change in trgen. I will also likely need a way to say "use these delimiter characters" for trgen, rather than hardwire to use some other delimiter characters for .csproj files.

Once I get all these Dependabot updates implemented, the build will be fabulous! Updates to new versions of Antlr will be a breeze.

**Comment by kaby76** [comment]
The reason is it not updated is exactly because the dependabot-core "nuget" parser cannot parse the .csproj file in template form. The solution is to create a "slave" .csproj, which must not contain StringTemplate escapes, then import the file in the "main" .csproj.
```
<Project>
  <ItemGroup>
    <PackageReference Include="Antlr4.Runtime.Standard" Version ="4.12.0" />
    <PackageReference Include="Antlr4BuildTasks" Version="12.2" PrivateAssets="all" />
  </ItemGroup>
</Project>
```
The main template, still in template form, requires this line to replace the `<ItemGroup>` for the package references: `\<Import Project="Other.csproj">\</Import>`. All the dotnet calls must explicitly reference the "main" .csproj, otherwise dotnet will be confused because there are two .csproj files.

**Comment by KvanTTT** [comment]
BTW, I think the file [Test.csproj](https://github.com/antlr/grammars-v4/blob/4ee2f74c11722d0b7d6c52cc6a86d7f2564fe11c/_scripts/templates/CSharp/Test.csproj) should be renamed to `Test.csproj.stg` because in fact it's more like template than csproj.

**Comment by kaby76** [comment]
> BTW, I think the file [Test.csproj](https://github.com/antlr/grammars-v4/blob/4ee2f74c11722d0b7d6c52cc6a86d7f2564fe11c/_scripts/templates/CSharp/Test.csproj) should be renamed to `Test.csproj.stg` because in fact it's more like template than csproj.

Good point. All files are processed using [Template](https://github.com/kaby76/Domemtech.StringTemplate4/blob/99eca35d124d2795522ae13d06d42a7f6dc38322/Antlr4.StringTemplate/Template.cs#L62), not [TemplateGroup](https://github.com/kaby76/Domemtech.StringTemplate4/blob/99eca35d124d2795522ae13d06d42a7f6dc38322/Antlr4.StringTemplate/TemplateGroup.cs#L79) right now. Using .stg's will be the better way, but it's more work than the simple solution I already implemented to get past this problem. I can add .stg processing in the next one or two version of trgen.

The plan is to work past the Dependabot problems so updating will be easier. I wrote a bug report to Github to clarify what they intend to do with updated releases of [dependabot-core](https://github.com/dependabot/dependabot-core). The processing of dotnet-tools.json files was implemented two months ago, but github isn't using the latest code.

Then, in theory, I can update trgen with .stg processing and have Dependabot just create a PR for the updated tool.
